From b317b393bf1ae3ae47afeaa15cf8831a8d62fc29 Mon Sep 17 00:00:00 2001
From: Yang Chen <chenyang@cs.utah.edu>
Date: Sun, 13 Apr 2014 01:43:56 -0700
Subject: [PATCH 08/27] fixed one more crash

if callee's type is of ElaboratedType, get its named type
rather than invoking getCallReturnType
---
 clang_delta/RemoveNestedFunction.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/clang_delta/RemoveNestedFunction.cpp b/clang_delta/RemoveNestedFunction.cpp
index 9ecf87f..5f86677 100644
--- a/clang_delta/RemoveNestedFunction.cpp
+++ b/clang_delta/RemoveNestedFunction.cpp
@@ -389,6 +389,10 @@ bool RemoveNestedFunction::addNewTmpVariable(ASTContext &ASTCtx)
     VarStr = DStr + " " + VarStr + ";";
     return RewriteHelper->addLocalVarToFunc(VarStr, TheFuncDecl);
   }
+  if (const ElaboratedType *ET = dyn_cast<ElaboratedType>(CalleeType)) {
+    QT = ET->getNamedType();
+    return writeNewTmpVariable(QT, VarStr);
+  }
   //  return writeNewIntTmpVariable(VarStr);
   QT = TheCallExpr->getCallReturnType();
   return writeNewTmpVariable(QT, VarStr);
-- 
2.0.4

